packages/app-project/package.json

```diff
@@ -6,9 +6,9 @@
   "version": "0.0.1",
   "main": "index.js",
   "scripts": {
-    "build": "PANOPTES_ENV=${PANOPTES_ENV:-production} next build",
+    "build": "APP_ENV=${APP_ENV:-development} PANOPTES_ENV=${PANOPTES_ENV:-production} next build",
     "build-storybook": "build-storybook -s public",
-    "dev": "PANOPTES_ENV=${PANOPTES_ENV:-staging} node server/dev-server.js",
+    "dev": "APP_ENV=${APP_ENV:-development} PANOPTES_ENV=${PANOPTES_ENV:-staging} node server/dev-server.js",
```

Comment:
I want to check my understanding of this change. If I run `master` locally, `console.log(process.env.APP_ENV)` inside `getDefaultPageProps()` returns APP_ENV as `production`. In this PR, doing the same returns `development`.

How does this change affect the following line in `getDefaultPageProps()`? And why was APP_ENV previously `production` when run locally?
https://github.com/zooniverse/front-end-monorepo/blob/b3b0ff1c600618defbc3c2c9abf56e439a5e79c8/packages/app-project/src/helpers/getDefaultPageProps/getDefaultPageProps.js#L13

Reply:
I saw that the default ironment envis `production` at the moment. I assumed that's a mistake. `HOSTS.production` is 'www.zooniverse.org', if I remember correctly, and that doesn't make sense, to me, for local development.

Reply:
Makes sense! When FEM is built for the staging site, APP_ENV is set at the following line, is that correct?
https://github.com/zooniverse/front-end-monorepo/blob/b3b0ff1c600618defbc3c2c9abf56e439a5e79c8/.github/workflows/deploy_staging.yml#L31
Then when FEM is built for production, APP_ENV is set here?
https://github.com/zooniverse/front-end-monorepo/blob/b3b0ff1c600618defbc3c2c9abf56e439a5e79c8/.github/workflows/ci-build.yml#L33

Reply:
Looks like it was originally set up just for staging, which explains why 'production' was the default: https://github.com/zooniverse/front-end-monorepo/pull/1529

Reply:
Here's the production build environment. Eventually, this will move from Jenkins to GitHub actions, like the staging build has already done.
https://github.com/zooniverse/front-end-monorepo/blob/b3b0ff1c600618defbc3c2c9abf56e439a5e79c8/Jenkinsfile#L69-L78

Reply:
Just to be on the safe side, I'm inclined to explicitly define `APP_ENV` in both kubernetes deploy templates, so that we aren't relying on defaults for the production deploys. If I do that, then it should be safe to change the default to 'development' in the code.

Reply:
I've updated the k8s templates, so I think `APP_ENV` is now explicitly set for both staging and production builds and deploys

Reply:
If you're wondering about the technical details, my fairly basic understanding of our deploy process is that there are two steps:
1. The Dockerfile builds an image that runs the `yarn bootstrap` script, so basically `yarn install` and `yarn build` for the libraries and the NextJS apps.
2.  Jenkins (production) or GitHub actions (staging) uses kubernetes to deploy that image in containers for each of the NextJS apps. I _think_  both NextJS apps use the same Docker image. The container runs `yarn start` for the already built NextJS app.

That's why we've got environment variables defined in a few different places: the Jenkinsfile, A GitHub Actions workflow and the two kubernetes templates.

Reply:
One thing I'm not sure about is if `APP_ENV` needs to be set for both `yarn build` and `yarn start`, which is why it's set in so many places.

Reply:
I think that's a good choice too.

Reply:
Ope Github didn't refresh the thread - I think that's a good choice in regards to explicitly setting APP_ENV in the k8s templates.